instrumentation/opentelemetry-instrumentation-aio-pika/src/opentelemetry/instrumentation/aio_pika/aio_pika_instrumentor.py

```diff
@@ -0,0 +1,46 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Collection
+
+import aio_pika
+
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+
+from .instrumented_exchange import (
+    InstrumentedExchange,
+    RobustInstrumentedExchange,
+)
+from .instrumented_queue import InstrumentedQueue, RobustInstrumentedQueue
+from .package import _instruments
+from .span_builder import SpanBuilder
+
+
+class AioPikaInstrumentor(BaseInstrumentor):
+    def _instrument(self, **kwargs):
+        tracer_provider = kwargs.get("tracer_provider", None)
+        SpanBuilder.TRACER_PROVIDER = tracer_provider
+        aio_pika.Channel.EXCHANGE_CLASS = InstrumentedExchange
+        aio_pika.Channel.QUEUE_CLASS = InstrumentedQueue
+        aio_pika.RobustChannel.EXCHANGE_CLASS = RobustInstrumentedExchange
+        aio_pika.RobustChannel.QUEUE_CLASS = RobustInstrumentedQueue
```

Comment:
Out of curiosity, why do you need the proxy objects and not just use `wrapt` like other instrumentations? This way if the user does this:
```python
channel = await connection.channel()
queue = await channel.declare_queue("hello")
AioPikaInstrumentor().instrument()
await channel.default_exchange.publish(......)
```
It will still instrument.

Reply:
the way i wrote it made it very easy, and i think that the usage of instrumentations would be in the beginning of the code anyway so i dont really see the added value of doing it this way

Reply:
Using `wrapt` aligns you with all other instrumentations, and opentelemetry already has the `unwrap` API maintained by them. I agree that this code works, but why manually monkey-patch when you have an API maintained by opentelemetry for that?

Reply:
got you, I read about `wrapt` a little bit and saw it used in other instrumentations, its pretty awesome 😄. 

changed 👍

Reply:
I know, right? 😄